api/src/main/java/io/minio/http/HttpUtils.java

```diff
@@ -126,6 +128,102 @@ public static String getHostHeader(HttpUrl url) {
+      keyStore.load(keyInput, keyStorePassword.toCharArray());
+    }
+    TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    trustManagerFactory.init(trustStore);
+
+    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+    keyManagerFactory.init(keyStore, keyStorePassword.toCharArray());
+
+    sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), new SecureRandom());
+    return sslContext.getSocketFactory();
+  }
+
+  private static X509TrustManager getX509TrustManager(String truststorePath, String trustStorePassword)
+          throws GeneralSecurityException, IOException {
+      KeyStore trustStore = KeyStore.getInstance("JKS");
+      try (InputStream trustInput = new FileInputStream(truststorePath)) {
+          trustStore.load(trustInput, trustStorePassword.toCharArray());
+      }
+      TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+      trustManagerFactory.init(trustStore);
+      return (X509TrustManager) trustManagerFactory.getTrustManagers()[0];
+  }
+
+    /**
+     * The client uses a certificate of the jks type.
+     *
+     * @param truststorePath Trust store file path
+     * @param keystorePath Key store file path
+     * @param trustStorePassword Trust store password
+     * @param keyStorePassword Key store password
+     * @return OkHttpClient
+     */
+  public static OkHttpClient newHttpClientByJKSCert(String truststorePath,
+                                                  String keystorePath,
+                                                  String trustStorePassword,
+                                                  String keyStorePassword){
+      OkHttpClient.Builder builder = new OkHttpClient.Builder();
+    if (truststorePath == null || keystorePath == null ||
+            truststorePath.isEmpty() || keystorePath.isEmpty()) {
+      return builder.build();
+    }
+    SSLSocketFactory sslSocketFactory = null;
+    X509TrustManager trustManager = null;
+    try {
+        sslSocketFactory = getSSLSocketFactory("JKS", truststorePath,
+                keystorePath, trustStorePassword, keyStorePassword);
+        trustManager = getX509TrustManager(truststorePath,trustStorePassword);
+    } catch (Exception e) {
+        throw new RuntimeException(e);
+    }
+    return builder.sslSocketFactory(sslSocketFactory, trustManager).build();
+  }
+
+
+    /**
+     * The client uses a certificate of the p12 type.
+     *
+     * @param truststorePath Trust store file path
+     * @param keystorePath Key store file path
+     * @param trustStorePassword Trust store password
+     * @param keyStorePassword Key store password
+     * @return OkHttpClient
+     */
+  public static OkHttpClient newHttpClientByP12Cert(String truststorePath,
```

Comment:
We don't want to create new http clients. Instead we would need to enable external certs. Over all this PR should be like below
```java
  private static OkHttpClient enableJKSPKCS12Certificates(
      OkHttpClient httpClient,
      String trustStorePath,
      String trustStorePassword,
      String keyStorePath,
      String keyStorePassword,
      boolean pkcs12)
      throws GeneralSecurityException, IOException {
    if (trustStorePath == null || trustStorePath.isEmpty()) {
      throw new IllegalArgumentException("trust store path must be provided");
    }
    if (trustStorePassword == null) {
      throw new IllegalArgumentException("trust store password must be provided");
    }
    if (keyStorePath == null || keyStorePath.isEmpty()) {
      throw new IllegalArgumentException("key store path must be provided");
    }
    if (keyStorePassword == null) {
      throw new IllegalArgumentException("key store password must be provided");
    }

    SSLContext sslContext = SSLContext.getInstance("TLS");
    KeyStore trustStore = KeyStore.getInstance("JKS");
    KeyStore keyStore = KeyStore.getInstance(pkcs12 ? "PKCS12" : "JKS");
    try (FileInputStream trustInput = new FileInputStream(truststorePath);
        FileInputStream keyInput = new FileInputStream(keystorePath); ) {
      trustStore.load(trustInput, trustStorePassword.toCharArray());
      keyStore.load(keyInput, keyStorePassword.toCharArray());
    }
    TrustManagerFactory trustManagerFactory =
        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
    trustManagerFactory.init(trustStore);

    KeyManagerFactory keyManagerFactory =
        KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
    keyManagerFactory.init(keyStore, keyStorePassword.toCharArray());

    sslContext.init(
        keyManagerFactory.getKeyManagers(),
        trustManagerFactory.getTrustManagers(),
        new java.security.SecureRandom());

    return httpClient
        .newBuilder()
        .sslSocketFactory(
            sslContext.getSocketFactory(),
            (X509TrustManager) trustManagerFactory.getTrustManagers()[0])
        .build();
  }

  public static OkHttpClient enableJKSCertificates(
      OkHttpClient httpClient,
      String trustStorePath,
      String trustStorePassword,
      String keyStorePath,
      String keyStorePassword)
      throws GeneralSecurityException, IOException {
    return enableJKSPKCS12Certificates(
        httpClient, trustStorePath, trustStorePassword, keyStorePath, keyStorePassword, false);
  }

  public static OkHttpClient enablePKCS12Certificates(
      OkHttpClient httpClient,
      String trustStorePath,
      String trustStorePassword,
      String keyStorePath,
      String keyStorePassword)
      throws GeneralSecurityException, IOException {
    return enableJKSPKCS12Certificates(
        httpClient, trustStorePath, trustStorePassword, keyStorePath, keyStorePassword, true);
  }
```